src/cargo/ops/vendor.rs

```diff
@@ -121,20 +152,31 @@ fn sync(
             .context("failed to download packages")?;
 
         for pkg in resolve.iter() {
+            let sid = if opts.respect_source_config {
+                source_replacement_cache.get(pkg.source_id())?
+            } else {
+                pkg.source_id()
+            };
+
             // Don't delete actual source code!
-            if pkg.source_id().is_path() {
-                if let Ok(path) = pkg.source_id().url().to_file_path() {
+            if sid.is_path() {
+                if let Ok(path) = sid.url().to_file_path() {
                     if let Ok(path) = try_canonicalize(path) {
                         to_remove.remove(&path);
                     }
                 }
                 continue;
             }
-            if pkg.source_id().is_git() {
+            if sid.is_git() {
                 continue;
             }
```

Comment:
nit: rather than `if`s with `continue`s that keep falling through cases, it seems like `if`/`else if` would better express the mutually exclusive nature of this loop.